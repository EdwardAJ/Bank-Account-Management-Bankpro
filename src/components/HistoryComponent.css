.wrapper {
    flex-wrap: wrap;
    display: flex;
    margin-top: 25px;
}