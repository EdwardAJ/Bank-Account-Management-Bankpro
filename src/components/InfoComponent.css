.info {
    color: white;
    line-height: 0.5rem;
}

.info p { 
    font-size: 0.85rem;
}

.info h3 {
    font-size: 1.2rem;
    margin-bottom: 2rem;
}