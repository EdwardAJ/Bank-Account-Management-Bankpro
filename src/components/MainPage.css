.container {
    display: block;
    height: 100vh;
}

.section-bottom {
    display: flex;
    font-family: 'Open Sans', sans-serif;
  }

.section-left { 
    width: 45%;
}

.section-top {
    height: 150px;
}

.menu {
    line-height: 1.75rem;
    margin-left: 80%;
    color: white;
    /* margin-bottom: 44.25%; */
    min-width: 500px;
}

.menu p {
    cursor: pointer;
    font-size: 0.85rem;
}

.active {
    font-weight: 700;
}

.menu h2 {
    font-size: 1.25rem;
}

.section-right {
    width: 55%;
    margin-left: 5%;
    max-width: 1000px;
}