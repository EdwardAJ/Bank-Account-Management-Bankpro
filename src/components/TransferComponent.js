import React from 'react';
import './TransferComponent.css';

import axios from 'axios';
import { TRANSFER_URL } from './../utils/endpoints.js';
import { HEADERS } from './../utils/header.js';
import getXMLResponse from './../utils/functions.js';
import { getCookie } from './../utils/cookie.js';

export default class TransferComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            destAccount: '',
            amount: ''
        };
        this.handleAmountChange = this.handleAmountChange.bind(this);
        this.handleDestAccountChange = this.handleDestAccountChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleAmountChange (event) {
        this.setState({amount: event.target.value});
    }

    handleDestAccountChange (event) {
        this.setState({destAccount: event.target.value});
    }

    handleSubmit (event) {
        var isVirtualAccount = false;
        if (this.state.destAccount.length > 8) {
            isVirtualAccount = true;
        }
        let requestBody =
        '<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:con="http://controllers.webservice.com/">' +
            '<soapenv:Header/>' +
            '<soapenv:Body>' +
                '<con:transferFund>' +
                    '<srcAccountNumber>' + getCookie('auth') + '</srcAccountNumber>' +
                    '<destAccountNumber>' + this.state.destAccount + '</destAccountNumber>' +
                    '<amount>' + this.state.amount + '</amount>' +
                    '<isVirtualAccount>' + isVirtualAccount + '</isVirtualAccount>' +
                '</con:transferFund>' +
            '</soapenv:Body>' +
        '</soapenv:Envelope>';

        axios.post(TRANSFER_URL, requestBody, {
            headers: HEADERS
        }).then((response) => {
            if (response.status === 200) {
                let resultStr = getXMLResponse(response.data)[0].innerHTML;
                alert(resultStr);
            }
        }).catch((error) => {
            alert("Error: ", error);
        });

        event.preventDefault();
    }

    render() {
        return (
            <div className="container-margin">
                <p className="login-header"> Nomor Rekening Transfer </p>
                <div className="form-area">
                    <form>
                        <input className="form" type="text" value={ this.state.destAccount } onChange={ this.handleDestAccountChange } placeholder="Example: 13517070" />
                    </form> 
                </div>
                <p className="login-header"> Jumlah Transfer </p>
                <div className="form-area">
                    <form>
                        <input className="form" type="text" value={ this.state.amount } onChange={ this.handleAmountChange } placeholder="Example: 40000" />
                    </form> 
                </div>
                <form>
                    <button className="submit" onClick={ this.handleSubmit }> TRANSFER </button>
                </form>
            </div>
        );
    }
}