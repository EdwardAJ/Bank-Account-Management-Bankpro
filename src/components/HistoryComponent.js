import React from 'react';
import './HistoryComponent.css';

import axios from 'axios';
import { HISTORY_URL } from './../utils/endpoints.js';
import { HEADERS } from './../utils/header.js';
import getXMLResponse from './../utils/functions.js';
import { getCookie } from './../utils/cookie.js';

export default class HistoryComponent extends React.Component {
    constructor(props) {
        super(props);
        // Request and get response from backend.
        this.fetchHistory();
    }

    fetchHistory() {
        let requestBody =
        '<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:con="http://controllers.webservice.com/">' +
            '<soapenv:Header/>' +
            '<soapenv:Body>' +
                '<con:getHistory>' +
                    '<accountNumber>' + getCookie("auth") + '</accountNumber>' +
                '</con:getHistory>' +
            '</soapenv:Body>' +
        '</soapenv:Envelope>';

        axios.post(HISTORY_URL, requestBody, {
            headers: HEADERS
        }).then((response) => {
            if (response.status === 200) {
                let key = 0;
                for (let i = 0; i < getXMLResponse(response.data).length; i++) {
                    if (i % 4 === 0) {
                        key++
                    }
                    getXMLResponse(response.data)[i].innerHTML
                }
            }
        }).catch((error) => {
            console.log("Error: ", error);
        })

    }
}
