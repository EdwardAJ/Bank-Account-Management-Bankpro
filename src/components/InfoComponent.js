import React from 'react';
import './InfoComponent.css';

import axios from 'axios';
import { INFO_URL } from './../utils/endpoints.js';
import { HEADERS } from './../utils/header.js';
import getXMLResponse from './../utils/functions.js';
import { getCookie } from './../utils/cookie.js';

export default class InfoComponent extends React.Component {
    constructor (props) {
        super(props);
        this.state = {
            nama: '',
            noRek: '',
            bank: '',
            saldo: ''
        };
        this.fetchUserInfo();
    }

    fetchUserInfo() {
        let requestBody =
        '<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:con="http://controllers.webservice.com/">' +
            '<soapenv:Header/>' +
            '<soapenv:Body>' +
                '<con:fetchUserInfo>' +
                    '<accountNumber>' + getCookie("auth") + '</accountNumber>' +
                '</con:fetchUserInfo>' +
            '</soapenv:Body>' +
        '</soapenv:Envelope>';

        axios.post(INFO_URL, requestBody, {
            headers: HEADERS
        }).then((response) => {
            if (response.status === 200) {
                let nama = getXMLResponse(response.data)[0].innerHTML;
                let noRek = getXMLResponse(response.data)[1].innerHTML;
                let bank = getXMLResponse(response.data)[2].innerHTML;
                let saldo = getXMLResponse(response.data)[3].innerHTML;
                this.setState({ nama: nama, noRek: noRek, bank: bank, saldo: saldo });
            }
        }).catch((error) => {
            alert("Error: ", error);
        });
    }

    render() {
        let nama = this.state.nama;
        let noRek = this.state.noRek;
        let bank = this.state.bank;
        let saldo = this.state.saldo;
        return (
            <div className="info">
                <p> Nasabah </p>
                <h3> { nama } </h3>
                <p> Nomor Rekening </p>
                <h3> { noRek } </h3>
                <p> Bank </p>
                <h3> { bank } </h3>
                <p> Saldo Terakhir </p>
                <h3> { saldo } </h3>  
            </div>
        );
    }
}