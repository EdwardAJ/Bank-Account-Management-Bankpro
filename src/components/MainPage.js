import React from 'react';
import './MainPage.css';
import InfoComponent from './InfoComponent';
import TransferComponent from './TransferComponent';
import HistoryComponent from './HistoryComponent';

export default class MainPage extends React.Component {
    constructor (props) {
        super(props);
        this.state = {
            choice: 1
        };
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick (choiceNum) {
        this.setState({choice: choiceNum});
    }

    render() {
        let choiceNum = this.state.choice;
        let component;
        if (choiceNum === 1) {
            component = <InfoComponent />;
        } else if (choiceNum === 2) {
            component = <TransferComponent />;
        } else if (choiceNum === 3) {
            component = <HistoryComponent />;
        }
        return (
            <div className="container">
                <div className="section-top">
                </div>
                <div className="section-bottom">
                    <div className="section-left">
                        <div className="menu">
                            <h2> bankpro </h2> 
                            <p id="id-1" className={ this.state.choice===1 ? 'active': 'default'} onClick={ () => this.handleClick(1) }> Home </p>
                            <p id="id-2" className={ this.state.choice===2 ? 'active': 'default'} onClick={ () => this.handleClick(2) }> Transfer </p>
                            <p id="id-3" className={ this.state.choice===3 ? 'active': 'default'} onClick={ () => this.handleClick(3) }> History </p>
                        </div>
                    </div>
                    <div className="section-right">
                        { component }
                    </div>
                </div>
            </div>
        );
    }
}

