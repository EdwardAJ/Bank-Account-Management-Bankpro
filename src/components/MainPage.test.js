import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';
import InfoComponent from './InfoComponent';
import TransferComponent from './TransferComponent';
import HistoryComponent from './HistoryComponent';

describe("MainPage", () => {
    it("should render my component", () => {
        const component = shallow(<MainPage />);
        expect(component.setState({ choice: 1 }));
        expect(component.find(InfoComponent)).toHaveLength(1);
        expect(component.find(TransferComponent)).toHaveLength(0);
        expect(component.find(HistoryComponent)).toHaveLength(0);

        expect(component.setState({ choice: 2 }));
        expect(component.find(InfoComponent)).toHaveLength(0);
        expect(component.find(TransferComponent)).toHaveLength(1);
        expect(component.find(HistoryComponent)).toHaveLength(0);

        expect(component.setState({ choice: 3 }));
        expect(component.find(InfoComponent)).toHaveLength(0);
        expect(component.find(TransferComponent)).toHaveLength(0);
        expect(component.find(HistoryComponent)).toHaveLength(1);
    });
});

describe("MainPage", () => {
    it("should simulate click", () => {
        const component = shallow(<MainPage />);
        expect(component.find('#id-2')).toHaveLength(1);
        component.find('#id-2').simulate('click');
        expect(component.state('choice')).toEqual(2);

        expect(component.find('#id-3')).toHaveLength(1);
        component.find('#id-3').simulate('click');
        expect(component.state('choice')).toEqual(3);

        expect(component.find('#id-1')).toHaveLength(1);
        component.find('#id-1').simulate('click');
        expect(component.state('choice')).toEqual(1);
    });
});