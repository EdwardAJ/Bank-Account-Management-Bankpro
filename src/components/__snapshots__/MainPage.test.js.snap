exports[`AuthPage should render my component 1`] = `
Array [
  <div
    className="container-auth">
    <div>
        <p
            id="login-header">
             Masukkan Nomor Rekening 
        </p>
        <div
            className="form-area">
            <form>
                <input
                    className="form"
                    onChange={[Function]}
                    placeholder="Example: 13517070"
                    type="text"
                    value="" />
            </form>
        </div>
        <form>
            <button
                className="submit"
                onClick={[Function]}>
                LOGIN
            </button>
        </form>
    </div>
</div>,
]
`;

exports[`MainPage should render my component 1`] = `
Array [
  <div
    className="container">
    <div
        className="section-top" />
    <div
        className="section-bottom">
        <div
            className="section-left">
            <div
                className="menu">
                <h2>
                     bankpro 
                </h2>
                <p
                    className="active"
                    onClick={[Function]}>
                     Home 
                </p>
                <p
                    className="default"
                    onClick={[Function]}>
                     Transfer 
                </p>
                <p
                    className="default"
                    onClick={[Function]}>
                     History 
                </p>
            </div>
        </div>
        <div
            className="section-right">
            <InfoComponent />
        </div>
    </div>
</div>,
]
`;
